src/server.go

```diff
@@ -236,17 +236,15 @@ func parseGetParams(query string) getParams {
 	for _, pair := range strings.Split(query, "&") {
 		parts := strings.SplitN(pair, "=", 2)
 		if len(parts) == 2 {
+			val, err := strconv.Atoi(parts[1])
+			if err != nil {
+				continue
+			}
 			switch parts[0] {
 			case "limit":
-				val, err := strconv.Atoi(parts[1])
-				if err == nil {
-					params.limit = val
-				}
+				params.limit = val
 			case "offset":
-				val, err := strconv.Atoi(parts[1])
-				if err == nil {
-					params.offset = val
-				}
+				params.offset = val
```

Comment:
I'm not sure about this. We don't want to call `Atoi` for other unsupported parameters.

This is what I'm suggesting.
```go
switch parts[0] {
case "limit", "offset":
	if val, err := strconv.Atoi(parts[1]); err == nil {
		if parts[0] == "limit" {
			params.limit = val
		} else {
			params.offset = val
		}
	}
}
```

Reply:
This looks much better! I'll update it as you suggested. Thank you for your feedback!